packages/mobile/src/fees/saga.ts

```diff
@@ -1,66 +1,101 @@
 import { getStableTokenContract } from '@celo/walletkit'
-import { call, put, select, spawn, takeLeading } from 'redux-saga/effects'
+import BigNumber from 'bignumber.js'
+import { call, put, takeLatest } from 'redux-saga/effects'
 import { showError } from 'src/alert/actions'
 import { ErrorMessages } from 'src/app/ErrorMessages'
-import { getReclaimEscrowFee } from 'src/escrow/saga'
+import { getReclaimEscrowGas } from 'src/escrow/saga'
 import { Actions, EstimateFeeAction, feeEstimated, FeeType } from 'src/fees/actions'
-import { getInvitationVerificationFee } from 'src/invite/saga'
-import { getSendFee } from 'src/send/saga'
+import { getInvitationVerificationFee, getInviteTxGas } from 'src/invite/saga'
+import { getSendTxGas } from 'src/send/saga'
 import { CeloDefaultRecipient } from 'src/send/Send'
+import { BasicTokenTransfer } from 'src/tokens/saga'
 import Logger from 'src/utils/Logger'
 import { web3 } from 'src/web3/contracts'
-import { currentAccountSelector } from 'src/web3/selectors'
+import { getGasPrice } from 'src/web3/gas'
+import { getConnectedAccount } from 'src/web3/saga'
 
 const TAG = 'fees/saga'
+// Cache of the gas estimates for common tx types
+// Prevents us from having to recreate txs and estimate their gas each time
+const feeGasCache = new Map<FeeType, BigNumber>()
+// Just use default values here since it doesn't matter for fee estimation
+const placeholderSendTx: BasicTokenTransfer = {
+  recipientAddress: CeloDefaultRecipient.address,
+  amount: web3.utils.fromWei('1'),
+  comment: 'Coffee or Tea?',
+}
 
-// TODO: skip fee update if it was calculated recently
 export function* estimateFeeSaga({ feeType }: EstimateFeeAction) {
   Logger.debug(`${TAG}/estimateFeeSaga`, `updating for ${feeType}`)
-
   try {
-    const account = yield select(currentAccountSelector)
+    const account = yield call(getConnectedAccount)
 
-    let feeInWei
+    let feeInWei: BigNumber | null = null
 
     switch (feeType) {
       case FeeType.INVITE:
-        feeInWei = yield call(getInvitationVerificationFee)
+        if (!feeGasCache.get(FeeType.INVITE)) {
+          const gas: BigNumber = yield call(
```

Comment:
As part of this PR I fixed Invite fees, which were previously not taking into account the cost of sending some money to escrow